setup.py

```diff
@@ -136,6 +136,15 @@ def run(self):
 
         SDist.run(self)
 
+        # Print warnings at the end because no one will see warnings before all the normal status
+        # output
+        if os.environ.get('_ANSIBLE_SDIST_FROM_MAKEFILE', False) != 1:
+            warnings.warn('When setup.py sdist is run from outside of the Makefile,'
```

Comment:
I'm worried that `pip install` users will see this and will keep reporting as a bug.

Maybe also check whether it's a Git checkout? Or maybe detect some env var RPM builder would usually set?

Reply:
If pip install creates an sdist from the existing sdist, then this whole strategy won't work and we will have to move the Makefile steps into the setup.py

Reply:
in my quick test, it looks like pip does not create an sdist so this is not a problem.  Do you concur?

Reply:
It seems like I'm being overly cautious. I've checked with @pradyunsg and he confirmed that pip isn't calling `sdist` command and it suppresses output anyway so users wouldn't see it unless in case of failure...